src/Overlay.tsx

```diff
@@ -40,6 +41,7 @@ const StyledOverlay = styled.div<StyledOverlayProps & SystemCommonProps & System
   min-width: 192px;
   max-width: 640px;
   height: ${props => heightMap[props.height || 'auto']};
+  max-height: ${props => props.maxHeight && heightMap[props.maxHeight]};
```

Comment:
The [`styled-components` docs](https://styled-components.com/docs/advanced#tagged-template-literals) explain:
> Speaking of which, during flattening, styled-components ignores interpolations that evaluate to undefined, null, false, or an empty string (""), which means you're free to use short-circuit evaluation to conditionally add CSS rules.

I verified this in Storybook: The styles for the `Overlay` in the “Single Select” story [exclude `max-height`](https://github.com/primer/components/blob/16d5568e9cddaa5257337df1aa30207d5db4473d/src/stories/SelectPanel.stories.tsx#L103); the styles for the `Overlay` in the “SelectPanel, Height: Initial, Overflowing Items” story [include `max-height`](https://github.com/primer/components/blob/16d5568e9cddaa5257337df1aa30207d5db4473d/src/stories/SelectPanel.stories.tsx#L134), as expected.

Reply:
Incidentally, I’m intentionally diverging from the [general guidance to prefer ternaries](https://github.com/primer/components/pull/1135#discussion_r614358653) here:
- because `props.maxHeight` has a fixed set of non-falsey possible values
- because both surrounding lines use logical operators for fallbacks
- because the rule is omitted in the falsey case, rather than set to a default value

Reply:
I'm good with this implementation.  I think the short circuit gotchas were very react-specific, so I'm fine with taking the easier approach if styled-components is more robust in what it accepts.